dev-util/clazy/patches/clazy-1.11.patchset

```diff
@@ -285,37 +285,179 @@ index b0da926..1ba1126 100644
+  * (method->getParent() doesn't need to equal record)
+@@ -107,7 +116,7 @@ inline bool classImplementsMethod(const clang::CXXRecordDecl *record, const clan
+ 
+     llvm::StringRef methodName = clazy::name(method);
+     for (auto m : record->methods()) {
+-        if (!m->isPure() && clazy::name(m) == methodName && parametersMatch(m, method))
++        if (!clazy::isPureVirtual(m) && clazy::name(m) == methodName && parametersMatch(m, method))
+             return true;
+     }
+ 
+diff --git a/src/checks/level2/virtual-call-ctor.cpp b/src/checks/level2/virtual-call-ctor.cpp
+index e6af81c..6bc2b34 100644
+--- a/src/checks/level2/virtual-call-ctor.cpp
++++ b/src/checks/level2/virtual-call-ctor.cpp
+@@ -23,6 +23,7 @@
+ */
+ 
+ #include "virtual-call-ctor.h"
++#include "FunctionUtils.h"
+ #include "HierarchyUtils.h"
+ #include "SourceCompatibilityHelpers.h"
+ #include "clazy_stl.h"
+@@ -115,7 +116,7 @@ SourceLocation VirtualCallCtor::containsVirtualCall(clang::CXXRecordDecl *classD
+             continue;
+ 
+         if (memberDecl->getParent() == classDecl) {
+-            if (memberDecl->isPure()) {
++            if (clazy::isPureVirtual(memberDecl)) {
+                 return clazy::getLocStart(callExpr);
+             } else {
+                 if (containsVirtualCall(classDecl, memberDecl->getBody(), processedStmts).isValid())
+-- 
+2.45.1
+
+
+From 44f32f88606abc72f7afeca18d51d5a9d43e9dbf Mon Sep 17 00:00:00 2001
+From: Cristian Adam <cristian.adam@gmail.com>
+Date: Mon, 12 Feb 2024 15:43:15 +0100
+Subject: Fix linker error with LLVM 18
+
+Change-Id: I4090a66912568b007619d6dacc2f91a9f5466767
+Reviewed-by: Alessandro Portale <alessandro.portale@qt.io>
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index d89325e..b503d13 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -118,6 +118,10 @@ if (${LLVM_VERSION} VERSION_GREATER_EQUAL "15.0.0")
+   set(clang_support_lib clangSupport)
+ endif()
+ 
++if (${LLVM_VERSION} VERSION_GREATER_EQUAL "18.0.0")
++  set(clang_api_notes_lib clangAPINotes)
++endif()
++
+ macro(link_to_llvm name is_standalone)
+   if (CLAZY_LINK_CLANG_DYLIB)
+     target_link_libraries(${name} clang-cpp)
+@@ -139,6 +143,7 @@ macro(link_to_llvm name is_standalone)
+     target_link_libraries(${name} clangTooling)
+     target_link_libraries(${name} clangToolingCore)
+     target_link_libraries(${name} ${clang_tooling_refactoring_lib})
++    target_link_libraries(${name} ${clang_api_notes_lib})
+   endif()
+ 
+   foreach(llvm_lib ${LLVM_LIBS})
+@@ -312,6 +317,7 @@ else()
+     clangStaticAnalyzerCheckers
+     clangStaticAnalyzerCore
+     clangStaticAnalyzerFrontend
++    ${clang_api_notes_lib}
+     )
+   add_executable(clazy-standalone ${CLAZY_STANDALONE_SRCS})
+ 
+-- 
+2.45.1
+
+
+From 6fd688c72eda9a1ccf76ac697cc8735ebb7b70fd Mon Sep 17 00:00:00 2001
+From: Schrijvers Luc <begasus@gmail.com>
+Date: Thu, 13 Jun 2024 13:50:31 +0200
+Subject: Disable PCH for 32bit
```

Comment:
@korli this isn't stricktly needed for 64bit, if so I could move this patch out and only apply it on 32bit only.

Reply:
Could be done later if need be. :)